packages/drivers/local-driver/src/localDocumentServiceFactory.ts

```diff
@@ -55,41 +48,16 @@ export class LocalDocumentServiceFactory implements IDocumentServiceFactory {
 		logger?: ITelemetryBaseLogger,
 		clientIsSummarizer?: boolean,
 	): Promise<IDocumentService> {
-		ensureFluidResolvedUrl(resolvedUrl);
-		if (createNewSummary === undefined) {
-			throw new Error("Empty file summary creation isn't supported in this driver.");
-		}
-		const pathName = new URL(resolvedUrl.url).pathname;
-		const pathArr = pathName.split("/");
-		const tenantId = pathArr[pathArr.length - 2];
-		const id = pathArr[pathArr.length - 1];
 		if (!this.localDeltaConnectionServer) {
 			throw new Error("Provide the localDeltaConnectionServer!!");
 		}
-		const documentStorage = (this.localDeltaConnectionServer as LocalDeltaConnectionServer)
-			.documentStorage;
-
-		if (!isCombinedAppAndProtocolSummary(createNewSummary)) {
-			throw new Error("Protocol and App Summary required in the full summary");
+		if (createNewSummary !== undefined) {
+			await createDocument(
+				this.localDeltaConnectionServer,
+				resolvedUrl as IFluidResolvedUrl,
```

Comment:
we should avoid casting. in this case, call ensureFluidResolvedUrl(resolvedUrl ) right before createDocument.